command/agent/cache/handler.go

```diff
@@ -20,11 +20,16 @@ import (
 	"github.com/hashicorp/vault/sdk/logical"
 )
 
-func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, inmemSink sink.Sink) http.Handler {
+func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, inmemSink sink.Sink, proxyVaultToken bool) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		logger.Info("received request", "method", r.Method, "path", r.URL.Path)
 
-		token := r.Header.Get(consts.AuthHeaderName)
+		token := ""
+		if proxyVaultToken {
+			token = r.Header.Get(consts.AuthHeaderName)
+		} else {
+			r.Header.Del(consts.AuthHeaderName)
+		}
```

Comment:
Minor nit: we can simplify this some by conditionally deleting the header and then always reading it for the value of `token`:

```go
if !proxyVaultToken {
	r.Header.Del(consts.AuthHeaderName)
}

token := r.Header.Get(consts.AuthHeaderName)
```

Avoids the `if/else` and is a minor simplification, but don't take it as a required change 😄